src/stores/notifications/RoomThreadsNotificationState.ts

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { NotificationColor } from "./NotificationColor";
+import { NotificationCountType } from "matrix-js-sdk/src/@types/receipt";
+import * as RoomNotifs from '../../RoomNotifs';
+import * as Unread from '../../Unread';
+import { RoomNotificationState } from "./RoomNotificationState";
+
+export class RoomThreadsNotificationState extends RoomNotificationState {
```

Comment:
the amount of copy/paste in this class is uncomfortable - why can't `RoomNotificationState` be used?

Reply:
There's a fair amount of duplicated code and I'll probably give a shot at DRYing it up

However there's some slight differences in how things are counted and cases that are relevant for the room level badge, but that aren't relevant for the threads.